### PR TITLE
CMCL-876: Confiner2D confines to midpoint when camera window is bigger than the axis aligned bounding box of the input confiner

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR

https://jira.unity3d.com/browse/CMCL-876
Confiner2D confines to midpoint when camera window is bigger than the axis aligned bounding box of the input confiner.

This case was not cover, and was reported by a user. + code cleanup

### Testing status

[Explanation of what’s tested, how tested and existing or new automation tests. Can include manual testing by self and/or QA. Specify test plans. Rarely acceptable to have no testing.]

- [ ] Added an automated test
- [x] Passed all automated tests
- [x] Manually tested 

### Documentation status

[Overview of how documentation is affected by this change. If there is no effect on documentation, explain why. Otherwise, state which sections are changed and why.]

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Technical risk

[Overall product level assessment of risk of change. Need technical risk & halo effect.]

### Comments to reviewers

[Info per person for what to focus on, or historical info to understand who have previously reviewed and coverage. Help them get context.]

### Package version

[Justification for updating either the patch, minor, or major version according to the [semantic versioning](https://semver.org/spec/v2.0.0.html) rules]

- [ ] Updated package version

### CODE REVIEW

File: com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs [hunk 1]
```diff
@@ -370,27 +370,17 @@ private void CalculateDeltaTransformationMatrix()
...
-        }
-
-        internal bool ConfinerOvenTimedOut()
-        {
-            return m_shapeCache.m_confinerOven != null 
-                && m_shapeCache.m_confinerOven.State == ConfinerOven.BakingState.TIMEOUT;
-        }
-    #endif
+        internal float BakeProgress() => m_shapeCache.m_confinerOven != null ? m_shapeCache.m_confinerOven.bakeProgress : 0f;
+        internal bool ConfinerOvenTimedOut() => m_shapeCache.m_confinerOven is { State: ConfinerOven.BakingState.TIMEOUT };
```
Comment (glabute) [inline com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs:374]: is a null check required?  I'm unfamiliar with this syntax

**Review by glabute — COMMENTED** [review]
I'm going to need you to explain the problem and the relevant logic changes.  I'm lost in all the cosmetic changes.

File: com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs (hunk 1 above)
Comment (gaborkb) [reply com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs:374]: This `m_shapeCache.m_confinerOven is { State: ConfinerOven.BakingState.TIMEOUT };` is equivalent to
`m_shapeCache.m_confinerOven != null && m_shapeCache.m_confinerOven.State == BakingState.TIMEOUT`

It is new to me too, if you prefer, we can revert it to the more verbose normal syntax.

File: com.unity.cinemachine/Runtime/Core/ConfinerOven.cs [hunk 2]
```diff
@@ -281,12 +252,26 @@ public Vector2 Unstretch(Vector2 p)
...
+                : Mathf.Min(m_Cache.userSetMaxFrustumHeight, frustumHeight);
+            
+            // Special case: we are shrank to the mid point of the original input confiner area.
+            if (State == BakingState.BAKED && frustumHeight > m_Cache.theoriticalMaxFrustumHeight)
+            {
+                return new BakedSolution(
+                    m_AspectStretcher.Aspect, frustumHeight, false,
+                    m_PolygonRect, m_OriginalPolygon, 
+                    new List<List<IntPoint>>{new List<IntPoint> { m_MidPoint }});
+            }
```
Comment (gaborkb) [inline com.unity.cinemachine/Runtime/Core/ConfinerOven.cs:267]: This is the new thing in the PR. 

What it does is that if the camera window is bigger (in both width and height) then the confining area. Then confine the camera to the mid point of the confining area.

File: com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs (hunk 1 above)
Comment (glabute) [reply com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs:374]: I find that it's a bit opaque.  Unless there's a specific advantage to using it, the more explicit code is better and more readable imo.

**Review by glabute — APPROVED** [review]

**Pull request merged**